src/App.tsx

```diff
@@ -1,26 +1,171 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header';
+import { ErrorNotification } from './components/ErrorNotification';
+import { FilterTodo } from './types/FilterTodo';
+import { Todo } from './types/Todo';
+import * as api from './api/todos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [errorMsg, setErrorMsg] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [adding, setAdding] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filtered, setFiltered] = useState<FilterTodo>('All');
+  const [deletedIds, setDeletedIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<null | Todo>(null);
+  const [updatingIds, setUpdatingIds] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    setLoading(true);
+
+    api
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMsg('Unable to load todos');
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    let filteredTodos = todos;
+
+    switch (filtered) {
+      case 'Active':
+        filteredTodos = filteredTodos.filter(td => !td.completed);
+        break;
+      case 'Completed':
+        filteredTodos = filteredTodos.filter(td => td.completed);
+        break;
+      default:
+        break;
+    }
+
+    return filteredTodos;
+  }, [filtered, todos]);
+
+  const changeVisibleTodos = (el: FilterTodo) => {
+    setFiltered(el);
+  };
+
+  async function addTodo(title: string) {
```

Comment:
It is important to stick to one way of creating functions everywhere
```suggestion
  const addTodo = async (title: string) => {
```